src/main/java/seedu/address/logic/parser/AddInternshipCommandParser.java

```diff
@@ -49,11 +49,20 @@ public AddInternshipCommand parse(String args) throws ParseException {
                 ParserUtil.parseInternshipRole(argMultimap.getValue(PREFIX_INTERNSHIP_ROLE).get());
         InternshipStatus internshipStatus =
                 ParserUtil.parseInternshipStatus(argMultimap.getValue(PREFIX_INTERNSHIP_STATUS).get());
-        PersonId contactPersonId = ParserUtil.parsePersonId(argMultimap.getValue(PREFIX_PERSON_ID).orElse(null));
-        InterviewDate interviewDate =
-                ParserUtil.parseInterviewDate(argMultimap.getValue(PREFIX_INTERVIEW_DATE).orElse(null));
 
-        return new AddInternshipCommand(companyName, internshipRole, internshipStatus, contactPersonId, interviewDate);
+        String interviewDateStr = argMultimap.getValue(PREFIX_INTERVIEW_DATE).orElse(null);
+        InterviewDate interviewDate = null;
+        if (interviewDateStr != null) {
+            interviewDate = ParserUtil.parseInterviewDate(interviewDateStr);
+        }
+
+        String linkIndexStr = argMultimap.getValue(PREFIX_LINK_INDEX).orElse(null);
```

Comment:
```suggestion
        String linkIndexStr = argMultimap.getValue(PREFIX_LINK_INDEX).orElse("");
```
Should this be an empty String instead of `null` instead? I'm not sure if there might be a `NullPointerException` somewhere.